### PR TITLE
Add support for kubernetes nodeSelector, tolerations and affinity

### PR DESCRIPTION
**Pull request by skinlayers**
##### SUMMARY
Add support for kubernetes nodeSelector, tolerations and affinity to installer.
I've also bumped the version of the installed postgresql helm chart from 2.0.0 to 3.18.3.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - Installer

##### AWX VERSION

```
awx: 4.0.0
```

##### ADDITIONAL INFORMATION
Nodes in a kubernetes cluster may be assigned "taints", to ensure only pods with matching "tolerations" may be scheduled to run on them. This is useful in shared kubernetes clusters, where different teams or applications may have a separate, dedicated pools of nodes.

See https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/

Additionally, pods may also have nodeSelectors that cause them to attract to nodes with matching labels.

See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector

Affinity is the successor to nodeSelector, and will eventually replace it.

See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity

I've moved the templating of values for the postgresql helm chart to an external template, so an arbitrary number of nodeSelector, tolerations and affinity entries may be added via jinja2 for loops. Unfortunately, defining multiple nodeSelector/tolerations/affinity entries via the inventory file is ugly. Feel free to suggest ways to make this more elegant:
```
node_selector={'organization':'foo', 'team':'bar'}
tolerations=[{'key':'organization', 'value':'foo', 'effect':'NoSchedule'}, {'key':'team', 'value':'bar', 'effect':'NoSchedule'}]
```
Alternatively, the tolerations & nodeSelector could be defined in a separate JSON or YAML file:
```
tolerations:
- key: "organization"
  value: "foo"
  effect: "NoSchedule"
- key: "team"
  value: "bar"
  effect: "NoSchedule"
nodeSelector:
  organization: foo
  team: bar
```

...and included with something like `--extra-vars '@tolerations.yml'`.

### CODE REVIEW

File: installer/roles/kubernetes/tasks/main.yml [hunk 1]
```diff
@@ -78,20 +78,18 @@
     - postgres_svc_details is defined and postgres_svc_details.rc != 0
     - openshift_host is defined
 
+- name: Render Postgres Values (Kubernetes)
+  set_fact:
+    postgres_values: "{{ lookup('template', 'postgresql-helm-values.yml.j2')"
```
Comment (shanemcd) [inline installer/roles/kubernetes/tasks/main.yml]: Hi and thanks for the PR. I believe we may need to wrap this conditional to ensure that it only runs on vanilla (non-openshift) Kubernetes. The below invocation of `shell` does this, so one option would be to move that condition to a `block` and wrap this new code with the proceeding task.

**Review by shanemcd — CHANGES_REQUESTED** [review]
I'm fine with this, just one small issue I noticed. Thanks again.

File: installer/roles/kubernetes/tasks/main.yml (hunk 1 above)
Comment (skinlayers) [reply installer/roles/kubernetes/tasks/main.yml]: Good catch! Change submitted along with a templating fix in latest commit.

**Comment by skinlayers** [comment]
@shanemcd This PR should be ready for review.

**Comment by skinlayers** [comment]
Spoke too soon. In my testing, the tolerations and nodeSelector aren't showing up on the postgresql pod. I'll do some more debugging...

**Comment by skinlayers** [comment]
@shanemcd Are there any remaining reasons this can't be merged yet?
Thank you!
EDIT: Let me do a test deployment in GKE today, and I'll reply with the results. I want to make sure that upgrading to the latest postgres helm chart doesn't break anything.

**Comment by skinlayers** [comment]
@shanemcd I've rebased on top of the latest devel as well as 6.1.0, and was able to deploy to GKE.

In my current deployment, I have an GCP instance running haproxy (dockerized) + certbot-google-dns. This gives me TLS, and allows me to hit AWX from anywhere inside our network (GCP ILBs can only be reached from within the same region, unless you put a proxy in front of it). `loadBalancerIP` is an internal address I had reserved, and `loadBalancerSourceRanges` is set to only allow the proxy instance to communicate with the ILB.

```
mkdir ansible
cd ansible
curl -o awx-6.1.0.tar.gz -L https://github.com/ansible/awx/archive/6.1.0.tar.gz
git clone https://github.com/ansible/awx-logos.git
tar xf awx-6.1.0.tar.gz
cd awx-6.1.0
python3 -m venv venv
. ./venv/bin/activate
pip install -U pip setuptools
pip install ansible docker
cd installer
cp inventory ../../my-inventory
```

Edit ../../my-inventory:
```
  kubernetes_context=gke_foo_cluster_context
  kubernetes_namespace=awx
  pg_password=CHANGME
  rabbitmq_password=CHANGEME_2
  rabbitmq_erlang_cookie=CHANGEME_3
  admin_password=CHANGEME_4
  create_preload_data=False
  secret_key=CHANGEME_5
  awx_official=true
```
Run installer:
```
ansible-playbook -i ../../my-inventory install.yml
```

my-ilb.yaml:
```
apiVersion: v1
kind: Service
metadata:
  name: awx-web-svc-ilb
  namespace: awx
  annotations:
    cloud.google.com/load-balancer-type: "Internal"
  labels:
    name: awx-web-svc-ilb
spec:
  type: LoadBalancer
  loadBalancerIP: X.X.X.X
  loadBalancerSourceRanges: ['Y.Y.Y.Y/32']
  selector:
    name: awx-web-deploy
  ports:
  - port: 80
    targetPort: 8052
    protocol: TCP
```
Post-install:
```
kubectl delete ingress.extension awx-web-svc
kubectl apply -f my-ilb.yaml
```

**Comment by skinlayers** [comment]
@ryanpetrello Is there anything else I can do to get this merged? I would rather not sit on this, and have to rebase again down the road.

**Comment by ryanpetrello** [comment]
@shanemcd you have any more feedback on this?  Are you ready to merge it?

**Review by shanemcd — APPROVED** [review]
My bad. My attention has been elsewhere recently and haven't had a chance to stand up a vanilla k8s cluster in a while. Looks good to me. Thanks for the contribution.

**Pull request merged**